### PR TITLE
java: Fix warnings in C code

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: bindings/java/c/evmc-vm.c [hunk 1]
```diff
@@ -78,9 +90,10 @@ JNIEXPORT void JNICALL Java_org_ethereum_evmc_EvmcVm_execute(JNIEnv* jenv,
...
                                                              jobject jmsg,
                                                              jobject jcode,
-                                                             jint jsize,
+                                                             jint jcode_size,
                                                              jobject jresult)
 {
+    (void)jcls;
     struct evmc_message* cmsg = (struct evmc_message*)(*jenv)->GetDirectBufferAddress(jenv, jmsg);
     assert(cmsg != NULL);
     const uint8_t* ccode = (uint8_t*)(*jenv)->GetDirectBufferAddress(jenv, jcode);
```
Comment (axic) [inline bindings/java/c/evmc-vm.c:99]: Maybe you want to remove the `c` prefixes from these things. And drop the large debugging comment?

File: bindings/java/c/host.c [hunk 2]
```diff
@@ -273,7 +279,8 @@ static size_t copy_code_fn(struct evmc_host_context* context,
                            uint8_t* buffer_data,
                            size_t buffer_size)
 {
-    size_t result;
+    (void)buffer_size;  // FIXME: buffer_size suspiciously unused.
```
Comment (axic) [inline bindings/java/c/host.c:282]: Ouch.

File: bindings/java/c/evmc-vm.c (hunk 1 above)
Comment (chfast) [reply bindings/java/c/evmc-vm.c:99]: Why to drop debugging section?

Comment (axic) [reply bindings/java/c/evmc-vm.c:99]: Is it useful? I don't mind.

**Pull request merged**